pkgs/native_assets_builder/test_data/simple_data_asset/hook/build.dart

```diff
@@ -1,18 +1,35 @@
 // Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
+import 'dart:io';
+
 import 'package:native_assets_cli/native_assets_cli.dart';
 
-void main(List<String> args) => build(
-      args,
-      (config, output) async {
+void main(List<String> args) async {
+  await build(
+    args,
+    (config, output) async {
+      final assetDirectory =
+          Directory.fromUri(config.packageRoot.resolve('assets/'));
+      // If assets are added, rerun hook.
+      output.addDependency(assetDirectory.uri);
+
+      await for (final dataAsset in assetDirectory.list()) {
```

Comment:
Do you want to add a `recursive: true` here and in other places?

Reply:
Flutter defaults to non-recursive. (It does directories, but not subdirectories.) But maybe we should at least add a param.